testlib/src/main/java/com/diffplug/spotless/StepHarness.java

```diff
@@ -43,7 +45,17 @@ public static StepHarness forStep(FormatterStep step) {
 						((FormatterStepImpl.Standard<?>) step).cleanupFormatterFunc();
 					}
 				},
-				input -> LineEnding.toUnix(step.format(input, new File("")))));
+				new FormatterFunc() {
+					@Override
+					public String apply(String input) throws Exception {
+						return LineEnding.toUnix(step.format(input, new File("")));
+					}
+
+					@Override
+					public String apply(String input, File source) throws Exception {
+						return LineEnding.toUnix(step.format(input, source));
+					}
+				}));
```

Comment:
This change should be reverted.  FormatterStep have to be able to handle `File source = null`, see my note on `ScalaFixStep`'s `FormatterFunc` for more info.

Reply:
Reverted.  I also updated the tests with `StepHarnessWithFile`, which is the appropriate way to test a step that needs the file on disk.